android/expoview/src/main/java/host/exp/exponent/experience/HomeActivity.java

```diff
@@ -64,7 +65,14 @@ protected void onCreate(Bundle savedInstanceState) {
 
     mSDKVersion = RNObject.UNVERSIONED;
     mManifest = mExponentManifest.getKernelManifest();
-    mExperienceId = ExperienceId.create(mManifest.optString(ExponentManifest.MANIFEST_ID_KEY));
+
+    String id;
+    try {
+      id = mManifest.getID();
```

Comment:
in a legacy manifest, the `id` field refers to the app/experience ID, whereas in a new manifest, the `id` field is a UUID referring to the update itself (and corresponds to the `releaseId` field in a legacy manifest).

i wonder if rather than having `getID` and `getReleaseID` on `LegacyRawManifest`, we should have `getExperienceID` (which actually points to `id`) and `getID` (which actually points to `releaseId`, since that's the ID we'd expect in other types of manifests). then here we can call `getExperienceID`. otherwise this and other places where we're expecting an experience ID might behave weirdly if they get an update ID with a new manifest at some point.

Reply:
This'll definitely be a good place to follow-up on, so thanks for pointing it out. For the scope of this PR though, I don't want to change behavior. In the subsequent PRs we'll fork the behavior of methods per type of manifest in accordance with the differing meaning of fields.